### ISSUE TITLE
position  embedding？ i cannot understand,the code?

### ISSUE THREAD

**Issue by henbucuoshanghai** [xiaohu2015/nngen#1]
half = dim // 2
    freqs = torch.exp(
        -math.log(max_period) * torch.arange(start=0, end=half, dtype=torch.float32) / half
    ).to(device=timesteps.device)
    args = timesteps[:, None].float() * freqs[None]
    embedding = torch.cat([torch.cos(args), torch.sin(args)], dim=-1)

**Comment by henbucuoshanghai** [comment]
can u explain it，tks

**Comment by xiaohu2015** [comment]
you can refer to transformer paper

**Comment by henbucuoshanghai** [comment]
whereis 10000?
